### PR TITLE
Lua workflow: Use Leafo's Luarocks action

### PR DESCRIPTION
**Pull request by appgurueu**
instead of installing outdated packages which lead to failing workflows

Observe that the checks complete successfully; without this PR the [luacheck & busted check fails](https://github.com/minetest/minetest/runs/6628264075?check_suite_focus=true).

### CODE REVIEW

File: .github/workflows/lua.yml [hunk 1]
```diff
@@ -42,9 +42,10 @@ jobs:
     runs-on: ubuntu-20.04
     steps:
     - uses: actions/checkout@v3
-    - name: Install luarocks
-      run: |
-        sudo apt-get update && sudo apt-get install -y luarocks
+    - uses: leafo/gh-actions-lua@v9
+      with:
+        luaVersion: "5.1.5"
```
Comment (sfan5) [inline .github/workflows/lua.yml:47]: I think you can skip this and install lua from the repos (`lua5.1`)

Comment (appgurueu) [reply .github/workflows/lua.yml:47]: Why would that be preferable?

Comment (sfan5) [reply .github/workflows/lua.yml:47]: might be faster, less third-party dependencies, ...

Comment (appgurueu) [reply .github/workflows/lua.yml:47]: Doesn't work; the luarocks action presumably depends on the Lua action.

Comment (sfan5) [reply .github/workflows/lua.yml:47]: smh

**Pull request merged**